### PR TITLE
feat: [sc-17206] ens_cure

### PR DESCRIPTION
**Pull request by Carbon225**
Story details: https://app.shortcut.com/ps-web3/story/17206

### CODE REVIEW

**Comment by Carbon225** [comment]
A lot of changes again, watch out for outdated docstrings, etc.

File: README.md [hunk 1]
```diff
@@ -47,62 +44,66 @@ ens_normalize('Nick.ETH')
...
 # ZWJ and '?' are removed, no error is raised
-# note: force conversion is not standardized
+# note: this function is not a part of the ENS Normalization Standard
 
-# note: might still raise DisallowedLabelError for certain names, which can not be force normalized, e.g.
-ens_force_normalize('abc..eth')
-# DisallowedLabelError: Contains a disallowed empty label
+# note: might still raise DisallowedNameError for certain names, which cannot be cured, e.g.
+ens_cure('abc..eth')
+# DisallowedNameError: Contains a disallowed empty label
```
Comment (djstrong) [inline README.md:100]: this works now

File: README.md [hunk 2]
```diff
@@ -167,64 +169,79 @@ try:
...
-| `FENCED_TRAILING` | Contains a disallowed character at the end of a label | This character is disallowed at the end of a label | ✅ |
-| `CONF_WHOLE` | Contains visually confusing characters that are disallowed | `None` | ❌ |
-| `CONF_MIXED` | Contains visually confusing characters from different scripts that are disallowed | This character is disallowed because it is visually confusing with another character from a different script | ✅ |
+## List of all `DisallowedNameError` types
+
+For fatal errors (not curable), it is challenging to communicate the normalization error as a problem with a specific substring.
+
+| `DisallowedNameErrorType` | General info |
+| ------------------------- | ------------ |
+| `EMPTY_NAME`   | The name is empty |
```
Comment (djstrong) [inline README.md:214]: is it curable now?

**Comment by Carbon225** [comment]
Now we have 2 errors for empty names/labels:
- `DisallowedNameError.EMPTY_NAME`: the input is empty, not curable
- `CurableErrorType.EMPTY_LABEL`: empty label detected (abc..eth), curable

**Comment by djstrong** [comment]
It is ready to merge, but I can't do it.

**Pull request merged**